src/Tizen.Multimedia.Vision/MediaVision/InferenceFaceDetector.cs

```diff
@@ -0,0 +1,208 @@
+
+        /// <summary>
+        /// Detects faces on the source image synchronously.
+        /// </summary>
+        /// <remarks>
+        /// <see cref="InferenceFaceDetectorResult.BoundingBoxes"/> can be empty, if there's no detected face.
+        /// </remarks>
+        /// <param name="source">The image data to detect faces.</param>
+        /// <returns>The BoundBoxes of detected face.</returns>
+        /// <exception cref="ObjectDisposedException">The InferenceFaceDetector already has been disposed.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        /// <since_tizen> 12 </since_tizen>
+        public InferenceFaceDetectorResult Inference(MediaVisionSource source)
+        {
+            ValidateNotDisposed();
+
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            InteropFD.Inference(_handle, source.Handle).Validate("Failed to inference face detection.");
+
+            return new InferenceFaceDetectorResult(_handle);
+        }
+
+        /// <summary>
+        /// Detects faces on the source image asynchronously.
+        /// </summary>
+        /// <remarks>
+        /// <see cref="InferenceFaceDetectorResult.BoundingBoxes"/> can be empty, if there's no detected face.<br/>
+        /// This method uses about twice as much memory as <see cref="InferenceFaceDetector.Inference"/>.
+        /// </remarks>
+        /// <param name="source">The image data to detect faces.</param>
+        /// <exception cref="ObjectDisposedException">The InferenceFaceDetector already has been disposed.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        /// <since_tizen> 12 </since_tizen>
+        public async Task<InferenceFaceDetectorResult> InferenceAsync(MediaVisionSource source)
+        {
+            ValidateNotDisposed();
+
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            InteropFD.InferenceAsync(_handle, source.Handle).Validate("Failed to inference face detection.");
+
+            return await Task.Factory.StartNew(() => new InferenceFaceDetectorResult(_handle),
+                CancellationToken.None,
+                TaskCreationOptions.DenyChildAttach | TaskCreationOptions.LongRunning,
+                TaskScheduler.Default);
+        }
+
+        private ulong _requestId = 1;
+        /// <summary>
+        /// Requests detecting faces to get their bounding boxes asynchronously.
+        /// </summary>
+        /// <remarks>
+        /// This function does not guarantee that inference is done when this method returns. The user can get the result by using <see cref="GetRequestResults"/>.<br/>
+        /// If the user calls this method again before the previous one is finished internally, the call will be ignored.<br/>
```

Comment:
would be nice if that situation was reflected in return value, but it seems that a new number will be returned anyway, so we don't have any additional information to put here